components/schemes/Color/src/MDCSemanticColorScheme.h

```diff
@@ -0,0 +1,138 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import "MDCColorScheme.h"
+
+/**
+ A simple color scheme that provides semantic context for the colors it uses. There are no optional
+ properties and all colors must be provided, supporting more reliable color theming.
+ */
+@protocol MDCColorScheming
```

Comment:
I suppose we don't technically need to implement NSObject here because all of the methods are required, but I want to confirm that the conformance was intentionally left out.

Reply:
I'm not sure, I tend not to force object requirements in protocols but I can add it. I have no feelings either way.

Reply:
Likely fine to leave as-is then until/if it becomes a problem :)

Reply:
Will clients need conformsToProtocol: if they are extending the protocol?

Reply:
Presumably if they are using conformsToProtocol: then they've already required a type that conforms to the NSObject protocol. Within our code and our themers I don't think we'll need to do any conformity or responder testing.

Reply:
SGTM.
We can add it if we need it.  Taking it away after we've added it, so much harder.